src/Compilers/CSharp/Portable/Symbols/NamespaceOrTypeSymbol.cs

```diff
@@ -325,6 +325,28 @@ internal virtual NamedTypeSymbol LookupMetadataType(ref MetadataTypeName emitted
                 }
             }
 
+            if (isTopLevel && MetadataHelpers.DecodeFileType(emittedTypeName.UnmangledTypeName) is (not -1 and var ordinal, var typeName))
```

Comment:
>if (isTopLevel && MetadataHelpers.DecodeFileType(emittedTypeName.UnmangledTypeName) is [](http://example.com/codeflow?start=12&length=86)

It looks like there is a couple of ```goto Done;``` above that jump over this ```if```. Is this what we want, or should we move the ```if``` below the label?

Reply:
I think it's more correct to still do this lookup even when `UseClsCompliantNameArityEncoding` is true and `InferredArity` or `ForcedArity` are positive. I'll move the 'if' below the label here.